src/main/java/com/muzika/homeworksb/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.muzika.homeworksb.service.impl;
+
+import com.muzika.homeworksb.dto.UserRegistrationRequestDto;
+import com.muzika.homeworksb.dto.UserRegistrationResponseDto;
+import com.muzika.homeworksb.enums.RoleEnum;
+import com.muzika.homeworksb.exception.RegistrationException;
+import com.muzika.homeworksb.mapper.UserMapper;
+import com.muzika.homeworksb.model.User;
+import com.muzika.homeworksb.model.UserRole;
+import com.muzika.homeworksb.repository.UserRepository;
+import com.muzika.homeworksb.repository.UserRoleRepository;
+import com.muzika.homeworksb.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+    private final UserRoleRepository userRoleRepository;
+
+    @Override
+    @Transactional(timeout = 1)
```

Comment:
if RegistrationException is checked exception, and you want to have a rollback behavior, than please use `rollbackFor` property